pandas/tests/indexes/test_indexing.py

```diff
@@ -96,3 +99,46 @@ def test_getitem_deprecated_float(idx):
 
     expected = idx[1]
     assert result == expected
+
+
+@pytest.mark.parametrize(
+    "idx, target, expected",
+    [
+        ([np.nan, "var1", np.nan], [np.nan], np.array([0, 2], dtype=np.intp)),
```

Comment:
does np.nan vs `float("NaN")` make a difference?

Reply:
I don't believe so. I changed all the np.nan to float("NaN") and tests are all passing. Would you like me to include a separate test case for float("NaN")?

Reply:
please add that testcase